docs/design/HotReloadDoc.md

```diff
@@ -0,0 +1,49 @@
+# Design Document: Hot Reloading for Runtime Configuration
+## Objective:
+The objective of this task is to enable "hot reloading" of the config file when we are not in a hosted scenario and we the Host Mode is set to `development`, in other words, in a local development scenario. To start we will implement this feature only working for the "runtime" section of the config file, but care must be made to implement this in a way that we do not block or have to change our implementation significantly for the other sections. This will allow users to update the configuration without the need to restart the service, providing more flexibility and efficiency within our application. After investigation it became apparent that hot reloading only 1 section of the config would be additional work without being able to make use of `IOptionsMonitor` to automatically update an individual section of the config that followed the `option` pattern. Because of the way we instantiate our object model of the configuration file, using `IOptionsMonitor`` in such a way was not possible at this time. We therefore will hot reload the entire config, but will lack the validation and implementation details for this to function inititally outside of Runtime section. Support for the other sections will be added incrementally, and when the entire hot reload feature is working, it will be documented for customer use.
```

Comment:
> > After investigation it became apparent that hot reloading only 1 section of the config would be additional work without being able to make use of `IOptionsMonitor` to automatically update an individual section of the config that followed the `option` pattern. Because of the way we instantiate our object model of the configuration file, using `IOptionsMonitor`` in such a way was not possible at this time.
> 
> I don't really follow. Are you saying you couldn't identify how to use `IOptionsMonitor` such that it only updates a single section of the config?

From my investigation, the `IOptionsMonitor` can not automatically reload the options from our config because of how we are creating our object model separately from a simply registering of the config file with the underlying .net framework. I was not able to get `IOptionsMonitor` to recognize or update when we change our config file in any of the ways that the documentation of tutorial describe.



> > We therefore will hot reload the entire config, but will lack the validation and implementation details for this to function inititally outside of Runtime section.
> 
> So the behavior of the first PR's worth of implementation will load a brand new file, but only use metadata from a single section (runtime) to rehydrate the configuration object maintained in the DAB engine at runtime?

So we will follow the same deserialization process as we do on startup, but without the ability to for example modify the schema hot reloading is not fully supported outside of the runtime section. Which is why we will not document this change until it is entirely supported.